src/cdk/testing/testbed/zone-types.d.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/*
+ * Type definitions for "zone.js". We cannot reference the official types
+ * using a triple-slash types directive because the types would bring in
+ * the NodeJS types into the compilation unit. This would cause unexpected
+ * type checking failures. We just create minimal type definitions for Zone
+ * here and use these for our interceptor logic.
+ */
+
+declare global {
+  // tslint:disable-next-line:variable-name
+  const Zone: {current: any}|undefined;
+}
+
+export type Zone = Object;
+export type ZoneDelegate = Object;
+export type HasTaskState = {microTask: boolean, macroTask: boolean};
+
+export interface ProxyZoneStatic {
```

Comment:
@devversion It looks like `ProxyZoneStatic` and `ProxyZone` are not part of the real zone.js `d.ts` file. Can you move these to a separate file (either `task-state-zone-interceptor.ts` or some new file like `proxy-zone-types.d.ts`). I need to be able to easily strip away just the ones that are part of the real zone.js `d.ts` file for google3.

Reply:
Makes sense. I've split the ProxyZone types into a separate file.